service/resource/endpoint/error.go

```diff
@@ -0,0 +1,20 @@
+package endpoint
+
+import (
+	"github.com/giantswarm/microerror"
+)
+
+var invalidConfigError = microerror.New("invalid config")
+var wrongTypeError = microerror.New("wrong type")
+var missingAnnotationError = microerror.New("annotation missing")
+
+// IsInvalidConfig asserts invalidConfigError.
+func IsInvalidConfig(err error) bool {
+	return microerror.Cause(err) == invalidConfigError
+}
+func IsWrongTypeError(err error) bool {
+	return microerror.Cause(err) == wrongTypeError
+}
+func MissingAnnotationError(err error) bool {
```

Comment:
I would really like to keep the error handling magic as we do it right now. Please see other projects how it is done. See also https://github.com/giantswarm/giantswarm/wiki/go-style-guide#errors. Concrete, I want to keep the error and its matcher together, the naming scheme of the matcher here is not aligned with our conventions and the error message of the error is "wrong".

Reply:
I cleaned this up, it should match the guideline now. Atleast from what I understood.